### ISSUE TITLE
MinimumTapTargetGuideline doesn't work if items are scrolled out of the viewport and the viewport doesn't touch the edge of the screen

### ISSUE THREAD

**Issue by xster** [flutter/flutter#107615]
b/238879865

MinimumTapTargetGuideline has logic to detect and skip elements that are scrolled partially off screen. However, this logic is flawed. It only checks for widgets that are at the edge of the physical screen, which omits the common case where the Scrollable container itself doesn't touch the edge of the screen.

Consider a screen with an AppBar and a long scrollable widget beneath it. Something like

```
Scaffold(
  appBar: AppBar(title: Text('Foo')),
  body: ListView(children: buildChildren()),
)
```

If a widget is partially visible at the top of the ListView, it will still be beneath the AppBar, not at the edge of the device screen, therefore it may fail the MinimumTapTargetGuideline erroneously.

**Comment by chunhtai** [comment]
repro

<details>

<summary>code</summary>

```dart
testWidgets('Accessibility requirements test', (WidgetTester tester) async {

    final SemanticsHandle handle = tester.ensureSemantics();
    final ScrollController controller = ScrollController();
    await tester.pumpWidget(
      MaterialApp(
        home: Scaffold(
          appBar: AppBar(title: Text('Foo')),
          body: ListView(
            controller: controller,
            children: [
              Padding(
                padding: EdgeInsets.only(left: 10, right: 10),
                child: SizedBox(width: 100, height: 100, child: Semantics( container: true, onTap: () {}, child: Text('hellp'))),
              ),
              Container(
                height: 1000,
                color: Colors.red,
              ),
            ]
          ),
        ),
      )
    );
    controller.jumpTo(90);
    await tester.pump();

    // debugDumpSemanticsTree(DebugSemanticsDumpOrder.traversalOrder);
    // debugDumpRenderTree();
    await expectLater(tester, meetsGuideline(iOSTapTargetGuideline));

    handle.dispose();

  });
```

</details>

There are also another problem where the check excluded target that has any of its bound touch the screen edge. I have to add padding to make sure the target is picked up by the check
